### PR TITLE
add filetransfer demo

### PR DESCRIPTION
**Pull request by fippo**
I was surprised by the lack of a filetransfer demo... let's provide one so people do not use examples that still use rtp data channels...

Performance seems to be between 15 and 20mbit/sec which is is somewhat lower than the 100mbps quoted in https://code.google.com/p/webrtc/issues/detail?id=2276 but it's a start.
This is actually quite interesting to watch with the new datachannel stats. Don't switch away during the filetransfer however. And remove the closeDataChannels call or get someone to merge https://code.google.com/p/chromium/issues/detail?id=416876 :-))

@jesup: TYMLTK -- works in firefox as well (setting .binaryType to arraybuffer)

### CODE REVIEW

**Comment by fippo** [comment]
looking at those progress bars in chrome and firefox is quite interesting actually. Firefox is sending way faster than Chrome (which is almost sync)

**Comment by samdutton** [comment]
Excellent – thank you! Will take a look.

File: src/content/datachannel/filetransfer/index.html [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  <link rel="stylesheet" href="css/main.css" />
+
+</head>
+
+<body>
+
+  <div id="container">
+
+    <h1><a href="https://googlechrome.github.io/webrtc/" title="WebRTC samples homepage">WebRTC samples</a> <span>RTCDataChannel Filetransfer</span></h1>
+
```
Comment (samdutton) [inline src/content/datachannel/filetransfer/index.html]: file transfer

File: src/content/datachannel/filetransfer/index.html [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  <link rel="stylesheet" href="../../../css/main.css">
+  <link rel="stylesheet" href="css/main.css" />
+
+</head>
+
+<body>
+
+  <div id="container">
+
+    <h1><a href="https://googlechrome.github.io/webrtc/" title="WebRTC samples homepage">WebRTC samples</a> <span>RTCDataChannel Filetransfer</span></h1>
```
Comment (samdutton) [inline src/content/datachannel/filetransfer/index.html]: Link should be to https://webrtc.github.io/samples/

(Just fixed this elsewhere.)

File: src/content/datachannel/filetransfer/index.html [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    <form id="fileInfo">
+      <input type="file" id="fileInput" name="files"/>
+    </form>
+
+    Send progress: <progress id="sendProgress" max="0" value="0"></progress>
```
Comment (samdutton) [inline src/content/datachannel/filetransfer/index.html]: Could you add a little more space (margin: 0 0 1em 0) between each or these lines? If you can be bothered, maybe put the progress elements on the same line as the label text, like http://webrtc.github.io/samples/src/content/getusermedia/volume.

File: src/content/datachannel/filetransfer/js/main.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+ */
+
+'use strict';
+
+var localConnection, remoteConnection, sendChannel, receiveChannel,
+  pcConstraint;
```
Comment (samdutton) [inline src/content/datachannel/filetransfer/js/main.js]: We've just started using JSCS validation with `"preset": "google"`. This adds a few validation rules such as only allowing one var declaration per line.

Could you add this and make the changes? There are less than 10, and they're all pretty trivial. 

FWIW there's a JSCS plugin for Sublime Text, if that helps.

**Comment by samdutton** [comment]
LGTM % nits.

Thanks again for doing this.

Will also ask Benjamin Schwartz to take a look.

**Comment by bemasc** [comment]
:+1:

**Comment by fippo** [comment]
All done... grunt seemed happy, but might be ignoring the file?

Now I just need to figure out what's the best way to merge the move of the basic sample with your changes. Probably rebase.

**Comment by fippo** [comment]
amazing, merged without problems. And grunt started being unhappy but that was easy to fix.

**Comment by samdutton** [comment]
LGTM

**Comment by fippo** [comment]
awesome.

Would be good to showcase the recently added datachannel stats there when https://c
ode.google.com/p/webrtc/issues/detail?id=4321 lands -- even though in the meantime the connection bitrate should be sufficient. I'll see if can do that as well, shouldn't be hard.

**Comment by juberti** [comment]
lgtm, cool!

**Pull request merged**